src/app/content/components/Topbar/index.tsx

```diff
@@ -35,191 +38,327 @@ interface Props {
-    }
-    return { ...state, queryProp: newProps.query };
-  }
+  return (
+    <Styled.MobileSearchWrapper mobileToolbarOpen={mobileToolbarOpen}>
+    <Styled.Hr />
+    <Styled.MobileSearchContainer>
+      {showBackToSearchResults &&
+        <FormattedMessage id='i18n:search-results:bar:toggle-text:mobile'>
+          {(msg) => <Styled.SeachResultsTextButton onClick={openSearchbar} data-testid='back-to-search-results'>
+            <Styled.LeftArrow /><Styled.InnerText>{msg}</Styled.InnerText>
+          </Styled.SeachResultsTextButton>}
+        </FormattedMessage>}
+      {!showBackToSearchResults &&
+        <FormattedMessage id='i18n:search-results:bar:close-text:mobile'>
+          {(msg) => <Styled.CloseSearchResultsTextButton onClick={toggleMobile} data-testid='close-search-results'>
+            <Styled.InnerText>{msg}</Styled.InnerText>
+          </Styled.CloseSearchResultsTextButton>}
+        </FormattedMessage>}
+      <Styled.SearchInputWrapper
+        action='#'
+        onSubmit={onSearchSubmit}
+        data-testid='mobile-search'
+        data-experiment
+        colorSchema={searchButtonColor}
+        searchInSidebar={searchInSidebar}
+      >
+        <Styled.SearchInput mobile type='search' data-testid='mobile-search-input'
+          autoFocus
+          onChange={onSearchChange} value={state.query} />
+        {!state.formSubmitted && !newButtonEnabled &&
+          <Styled.SearchButton desktop colorSchema={searchButtonColor} data-experiment />
+        }
+        {
+          state.query && newButtonEnabled && <Styled.CloseButtonNew
+            type='button'
+            aria-label='clear search'
+            onClick={onSearchClear}
+            formSubmitted={state.formSubmitted}
+            data-testid='mobile-clear-search'
+          >
+            <Styled.CloseIcon />
+          </Styled.CloseButtonNew>
+        }
+        {
+          state.query && !newButtonEnabled && <Styled.CloseButton
+            type='button'
+            aria-label='clear search'
+            onClick={onSearchClear}
+            formSubmitted={state.formSubmitted}
+            data-testid='mobile-clear-search'
+          />
+        }
+      </Styled.SearchInputWrapper>
+      {children}
+    </Styled.MobileSearchContainer>
+  </Styled.MobileSearchWrapper>
+
+  );
+}
+
+// Effectively a conditionally executed hook
+function AltSCycler({hasSearchResults}: {hasSearchResults: boolean}) {
+  const isMobile = useMatchMobileQuery();
+  const cycleSearchRegions = React.useCallback(
+    // Only in desktop layout
+    () => {
+      if (isMobile) {
+        return;
+      }
+      const targets = [
+        '[class*="SearchInputWrapper"] input',
+        '[class*="SearchResultsBar"]',
+        'main',
+      ].map((q) => document?.querySelector(q));
+
+      // Determine which region we are in (if any)
+      const currentSectionIndex = targets.findIndex((el) => el?.contains(document?.activeElement!));
```

Comment:
I'd do something like `document?.activeElement && el?.contains(document.activeElement)` because activeElement can be null and idk if contains likes nulls or undefineds.

Reply:
`null` is ok for `contains`. `activeElement` _can_ be undefined, but there's no good way to make it so. In general, it will be set to `body` if you `blur` it from anything else.
I changed `document?activeElement!` to `document?activeElement ?? null` but could not write a test to hit the undefined branch.